File: TS3-LogDataViewer/Ban.cpp
// Ban.cpp : Methods of the Ban class.
// Author : Drumsticks1
// GitHub : https://github.com/Drumsticks1/TS3-LogDataViewer

#include <vector>
#include <string>
#include "Ban.h"

std::vector <Ban> BanList;

// Sets the data of the current Ban object according to the given data.
void Ban::addBan(std::string banDateTime, unsigned int bannedID, std::string bannedNickname, std::string bannedUID, std::string bannedIP, std::string bannedByNickname, unsigned int bannedByID, std::string bannedByUID, std::string banReason, unsigned int bantime) {
	this->banDateTime = banDateTime;
	this->bannedID = bannedID;
	this->bannedNickname = bannedNickname;
	this->bannedUID = bannedUID;
	this->bannedIP = bannedIP;
	this->bannedByID = bannedByID;
	this->bannedByNickname = bannedByNickname;
	this->bannedByUID = bannedByUID;
	this->banReason = banReason;
	this->bantime = bantime;
}

// Returns the requested information.
std::string Ban::getBanDateTime() { return banDateTime; }
unsigned int Ban::getBannedID() { return bannedID; }
std::string Ban::getBannedNickname() { return bannedNickname; }
std::string Ban::getBannedUID() { return bannedUID; }
std::string Ban::getBannedIP() { return bannedIP; }
std::string Ban::getBannedByNickname() { return bannedByNickname; }
unsigned int Ban::getBannedByID() { return bannedByID; }
std::string Ban::getBannedByUID() { return bannedByUID; }
std::string Ban::getBanReason() { return banReason; }
unsigned int Ban::getBantime() { return bantime; }

File: nodejs/Ban.js
// Ban.js: Ban class.
// Author : Drumsticks
// GitHub : https://github.com/Drumsticks1/TS3-LogDataViewer

"use strict";

/**
 * @param {number} banListId
 * @param {string} banDateTime
 * @param {number} bannedID
 * @param {string} bannedNickname
 * @param {string} bannedUID
 * @param {string} bannedIP
 * @param {number} bannedByID
 * @param {string} bannedByNickname
 * @param {string} bannedByUID
 * @param {string} banReason
 * @param {number} banTime
 * @constructor
 */
const Ban = function (banListId, banDateTime, bannedID, bannedNickname, bannedUID, bannedIP, bannedByID, bannedByNickname, bannedByUID, banReason, banTime) {
  this.banListId = banListId;
  this.banDateTime = banDateTime;
  this.bannedID = bannedID;
  this.bannedNickname = bannedNickname;
  this.bannedUID = bannedUID;
  this.bannedIP = bannedIP;
  this.bannedByID = bannedByID;
  this.bannedByNickname = bannedByNickname;
  this.bannedByUID = bannedByUID;
  this.banReason = banReason;
  this.banTime = banTime;
};

// Returns the requested information.
Ban.prototype.getBanDateTime = function () {
  return this.banDateTime;
};
Ban.prototype.getBannedID = function () {
  return this.bannedID;
};
Ban.prototype.getBannedNickname = function () {
  return this.bannedNickname;
};
Ban.prototype.getBannedUID = function () {
  return this.bannedUID;
};
Ban.prototype.getBannedIP = function () {
  return this.bannedIP;
};
Ban.prototype.getBannedByNickname = function () {
  return this.bannedByNickname;
};
Ban.prototype.getBannedByID = function () {
  return this.bannedByID;
};
Ban.prototype.getBannedByUID = function () {
  return this.bannedByUID;
};
Ban.prototype.getBanReason = function () {
  return this.banReason;
};
Ban.prototype.getBanTime = function () {
  return this.banTime;
};

module.exports = {
  Ban: Ban,

  /**
   * Adds a new Ban with the given data to the array.
   * @param {Array} array
   * @param {string} banDateTime
   * @param {number} bannedID
   * @param {string} bannedNickname
   * @param {string} bannedUID
   * @param {string} bannedIP
   * @param {number} bannedByID
   * @param {string} bannedByNickname
   * @param {string} bannedByUID
   * @param {string} banReason
   * @param {number} banTime
   */
  addBan: function (array, banDateTime, bannedID, bannedNickname, bannedUID, bannedIP, bannedByID, bannedByNickname, bannedByUID, banReason, banTime) {
    array.push(
      new Ban(
        array.length + 1,
        banDateTime,
        bannedID,
        bannedNickname,
        bannedUID,
        bannedIP,
        bannedByID,
        bannedByNickname,
        bannedByUID,
        banReason,
        banTime
      ));
  }
};
